app/controllers/components_controller.rb

```diff
@@ -11,6 +12,7 @@ def index
 
   def preview
     @component = Component.get(params[:component_id])
+    head :not_found if @component.nil?
```

Comment:
In this instance we have two explicit options for rendering that are conditional on the component. The code would be easier to read using `if` `else` `end` like so:

```rb
if @component.nil?
  head :not_found
else
  render layout: 'preview'
end
```